apis/core/v1alpha1/flagsourceconfiguration_types.go

```diff
@@ -66,29 +73,66 @@ type FlagSourceConfigurationSpec struct {
 	// +optional
 	Evaluator string `json:"evaluator"`
 
-	// Image allows for the flagd image to be overridden, defaults to 'ghcr.io/open-feature/flagd'
+	// Image allows for the sidecar image to be overridden, defaults to 'ghcr.io/open-feature/flagd'
 	// +optional
 	Image string `json:"image"`
 
-	// Tag to be appended to the flagd image, defaults to 'main'
+	// Tag to be appended to the sidecar image, defaults to 'main'
 	// +optional
 	Tag string `json:"tag"`
 }
 
-func NewFlagSourceConfigurationSpec() *FlagSourceConfigurationSpec {
-	var tag = defaultTag
-	if flagDVersion := os.Getenv(flagDVersionEnvVar); flagDVersion != "" {
-		tag = flagDVersion
-	}
-	return &FlagSourceConfigurationSpec{
+func NewFlagSourceConfigurationSpec() (*FlagSourceConfigurationSpec, error) {
+	var parseError error = nil
+	fsc := &FlagSourceConfigurationSpec{
 		MetricsPort:      defaultMetricPort,
 		Port:             defaultPort,
 		SocketPath:       defaultSocketPath,
 		SyncProviderArgs: []string{},
 		Evaluator:        defaultEvaluator,
 		Image:            defaultImage,
-		Tag:              tag,
+		Tag:              defaultTag,
+	}
+
+	if metricsPort := os.Getenv(fmt.Sprintf("%s_%s", InputConfigurationEnvVarPrefix, SidecarMetricPortEnvVar)); metricsPort != "" {
+		metricsPortI, err := strconv.Atoi(metricsPort)
+		if err != nil {
+			parseError = err
+		} else {
+			fsc.MetricsPort = int32(metricsPortI)
+		}
+	}
+
+	if port := os.Getenv(fmt.Sprintf("%s_%s", InputConfigurationEnvVarPrefix, SidecarPortEnvVar)); port != "" {
+		portI, err := strconv.Atoi(port)
+		if err != nil {
+			parseError = err
```

Comment:
Why bother carry on if there's an error? Does this suggest that the error returned from this func isn't fatal to the `FlagSourceConfigurationSpec` construction?
If so, I'd suggest adding an explanation of this in the spec for this func